src/Http/Http/test/PipeWriterTests.cs

```diff
@@ -90,13 +90,11 @@ public void CanGetSameMemoryWhenNoAdvance()
         }
 
         [Fact]
-        public void CanGetNewSpanWhenNoAdvanceWhenSizeTooLarge()
+        public void GetSpanWithZeroSizeHintReturnsMaxBufferSizeOfPool()
         {
             var span = Writer.GetSpan(0);
 
-            var secondSpan = Writer.GetSpan(10000);
-
-            Assert.False(span.SequenceEqual(secondSpan));
+            Assert.Equal(4096, span.Length);
```

Comment:
Can you verify this same result for when 0 < sizeHint < 4096?

Can you also update CanGetNewMemoryWhenSizeTooLarge to verify that memoryLarge.Length is equal to `MemoryPool<byte>.MaxBufferSize`?